### ISSUE TITLE
Removing an empty directory not present in all disks fail to remove its parent directory although empty

### ISSUE THREAD

**Issue by vadmeste** [minio/minio#6529]
## Expected Behavior
When I remove an empty directory living alone in its parent directory, the parent directory should be removed in all cases.

## Current Behavior
When an empty directory is not present in all disks (not healthy), removing it won't remove the parent directory although the latter becomes empty

## Possible Solution
Do not know

## Steps to Reproduce (for bugs)
1. Start Minio standalone erasure with 4 fresh disks
2. aws s3api put-object --bucket testbucket --key dir/emptydir/ --endpoint http://minio:9010
3. Remove emptydir in two disks from the backend
4. mc rm myminio/testbucket/dir/emptydir/
5. mc ls myminio/testbucket/ and mc ls myminio/testbucket/dir/

## Context
Fixing an empty object issue

## Regression
No

## Your Environment
All

**Comment by vadmeste** [comment]
Basically this is the needed fix:

```diff
diff --git a/cmd/xl-v1-object.go b/cmd/xl-v1-object.go
index e705a109..831a0e48 100644
--- a/cmd/xl-v1-object.go
+++ b/cmd/xl-v1-object.go
@@ -507,7 +507,7 @@ func undoRename(disks []StorageAPI, srcBucket, srcEntry, dstBucket, dstEntry str
 
 // rename - common function that renamePart and renameObject use to rename
 // the respective underlying storage layer representations.
-func rename(ctx context.Context, disks []StorageAPI, srcBucket, srcEntry, dstBucket, dstEntry string, isDir bool, writeQuorum int, ignoredErr []error) ([]StorageAPI, error) {
+func rename(ctx context.Context, disks []StorageAPI, srcBucket, srcEntry, dstBucket, dstEntry string, isDir bool, writeQuorum int, ignoredErrs []error) ([]StorageAPI, error) {
 	// Initialize sync waitgroup.
 	var wg = &sync.WaitGroup{}
 
@@ -528,11 +528,7 @@ func rename(ctx context.Context, disks []StorageAPI, srcBucket, srcEntry, dstBuc
 		wg.Add(1)
 		go func(index int, disk StorageAPI) {
 			defer wg.Done()
-			if err := disk.RenameFile(srcBucket, srcEntry, dstBucket, dstEntry); err != nil {
-				if !IsErrIgnored(err, ignoredErr...) {
-					errs[index] = err
-				}
-			}
+			errs[index] = disk.RenameFile(srcBucket, srcEntry, dstBucket, dstEntry)
 		}(index, disk)
 	}
 
@@ -541,7 +537,7 @@ func rename(ctx context.Context, disks []StorageAPI, srcBucket, srcEntry, dstBuc
 
 	// We can safely allow RenameFile errors up to len(xl.getDisks()) - writeQuorum
 	// otherwise return failure. Cleanup successful renames.
-	err := reduceWriteQuorumErrs(ctx, errs, objectOpIgnoredErrs, writeQuorum)
+	err := reduceWriteQuorumErrs(ctx, errs, append(objectOpIgnoredErrs, ignoredErrs...), writeQuorum)
 	if err == errXLWriteQuorum {
 		// Undo all the partial rename operations.
 		undoRename(disks, srcBucket, srcEntry, dstBucket, dstEntry, isDir, errs)
```

reduceWriteQuorumErrs() considers 'nil' as an error and raise it if it satisfies quorum but rename() here sets errs[index] is nil if the error is actually not found, which misleads reduceWriteQuorumErrs

**Comment by harshavardhana** [comment]
Looks like a correct fix @vadmeste

**Comment by vadmeste** [comment]
This causes a problem, when an object is present in only half of disks, uploading a new object with the same name will fail. The reason is PutObject() tries to rename the existing object to tmp directory but the latter doesn't have a quorum

**Comment by harshavardhana** [comment]
This issue should be fixed by #6670

**Comment by harshavardhana** [comment]
This is not fixed by #6670

**Comment by vadmeste** [comment]
I think this issue can be closed, this scenario I created manually in my local machine because I thought someone was able to reproduce it.. but after some investigation, it turned out that the real reason is that the empty directory exists because it has some stales objects in it.

Closing this.
